packages/gensx-core/src/utils/project-config.ts

```diff
@@ -0,0 +1,51 @@
+import { readFileSync } from "node:fs";
+import path from "node:path";
+
+import { z } from "zod";
+
+// Define schema for gensx.yaml
+const ProjectConfigSchema = z.object({
+  projectName: z.string(),
+  environmentName: z.string().optional(),
+  description: z.string().optional(),
+});
+
+export type ProjectConfig = z.infer<typeof ProjectConfigSchema>;
+
+/**
+ * Get the path to the gensx.yaml file
+ */
+export function getProjectConfigPath(dir: string): string {
+  return path.join(dir, "gensx.yaml");
+}
+
+/**
+ * Read the gensx.yaml file and return the parsed config
+ */
+export function readProjectConfig(dir: string): ProjectConfig | null {
```

Comment:
I feel like we added support for JSON or yaml. Maybe that was a fever dream.

Reply:
this is identical to the code from the CLI (albeit sync instead of async). Looks like only yaml is supported.

Reply:
Ok. Hallucination then lol. Should this use the same code as the CLI?

Reply:
is there a recommended way to share between `gensx` and `gensx-core`? the other config stuff had just been duplicated between those two folders already but i agree it'd be nicer to only have the code in one place